packages/cli/templates/react/src/useFieldPlugin.ts

```diff
@@ -1,16 +1,17 @@
-import { createFieldPlugin, FieldPluginResponse } from '@storyblok/field-plugin'
-import { useEffect, useState } from 'react'
+import { FieldPluginResponse } from '@storyblok/field-plugin'
+import { useContext } from 'react'
+import { FieldPluginContext } from './FieldPluginProvider'
 
-type UseFieldPlugin = () => FieldPluginResponse
+export const useFieldPlugin = (): Extract<
+  FieldPluginResponse,
+  { type: 'loaded' }
+> => {
+  const plugin = useContext(FieldPluginContext)
+  if (!plugin) {
+    throw new Error(
+      'The plugin is not loaded, yet `useFieldPlugin()` was invoked. Ensure that the component that invoked `useFieldPlugin()` is wrapped within `<FieldPluginProvider>`.',
+    )
+  }
```

Comment:
I removed the second conditional where useFieldPlugin was checking `type !== 'loaded`, because this can never occur. I kept the error message in the first if-statement.

Reply:
Technically, it can. If `<Loading>` or `<Error>` component uses this hook, they will have different type. But I'm fine with this.